### ISSUE TITLE
Titlovi / Subtitles

### ISSUE THREAD

**Issue by kikipmtgf** [rsimac/kodi_plugin_hrti#5]
Hello,

First of all, thanks a lot for this plugin and sorry to write in English, my Croatian is still very poor.

I was wondering if at some point you could include an option to activate the subtitles provided by HRT on HRTi - I don't know if it is feasible or not, so this is really a feature request more than an issue! But it would help me follow some streams with my native Croatian wife :)

**Comment by rsimac** [comment]
Yeah that is definitely a valid request, subtitles are great help...

I checked the hrti website, picked one movie, but was not able to see the subtitle/cc option at their site... Where exactly did you observed the existing hrti subtitles? If you can point me to the name of movie/show so I can take a look into how to move this into kodi add-on.

Thanks!

**Comment by kikipmtgf** [comment]
Hi,

Thanks for your reply!

You can try an episode of Crno-bijeli svijet, for instance the last one: https://hrti.hrt.hr/video/show/2998391/crno-bijeli-svijet-ljudi-samoce-2-sezona-1212-hd-t

Here you can activate subtitles.

![image](https://user-images.githubusercontent.com/31631383/30071905-eecc0754-9268-11e7-9d12-e6600bdb3038.png)

**Comment by rsimac** [comment]
Update, did not forget about this issue, found the subtitle files and link, but was unable to relate them to the video... Busy with day job right now, but will look into this further... I like subtitles too...

**Comment by kikipmtgf** [comment]
Thank for the update! Where did you find the subtitle files? I can't find the link. Maybe we can load them manually in Kodi.

Good luck with your day job !

**Comment by rsimac** [comment]
For example, when analyzing network traffic,  the episode 12/12 of 'crno bijeli svijet'

has subtitles at:
https://hrti.hrt.hr/uploads/11/captions/152cef657bd6f2e96a4ace53b864c44f.vtt

but I was unable to relate the subtitle link (name, folder) to the actual video...

**Comment by kikipmtgf** [comment]
Good catch, thank you! I guess they may be using the referer as the key for the subtitles request, in which they have to store somewhere a matching table (that links URL and subtitles resources). It would not be good news as we probably cannot access directly this table...

**Comment by rsimac** [comment]
Good news and bad news. There is no server side 'table', the subtitle uri is visible in json catalog client (hrti) gets from server. Also, the Kodi is able to display the subtitles, I tried it by hardcoding the uri.

Bad news, the json catalog we get from server has (some) duplicate entries for old (no subtitles) and new (subtitled) videos... one containing subtitle uri, the other not, and them having different video id,  and this need some acrobatics to figure out... Stay tuned...

**Comment by rsimac** [comment]
I managed to implement the subtitles, tested working with 'crno bijeli svijet'.

However I had to introduce new network calls, now the data is being loaded for each episode as they are listed. This may slow down entire experience from previous O(1) to O(N) where N is the number of videos listed. I did not noticed the major slowdown yet, but for time being I will not merge these changes to master branch, and will think of alternate solution. Also will accept any suggestions...

For subtitle testing, feel free to use the dev branch, the addon zip can be downloaded from:
https://github.com/rsimac/kodi_plugin_hrti/blob/dev/plugin_video_hrti.zip

Do let me know how is the overall feel.

**Comment by muzena** [comment]
Ja sam probao, radi samo na Crno bijeli svijet, isprobavao sam na filmovima i neke serije, tamo ili nema titlova ili ne rade.

**Comment by rsimac** [comment]
Hvala na testu, moguce je da neke epizode ili filmovi nemaju titlove uopce. Kad budem imao vremena provjeriti cu na hrti.hr direktno...

**Comment by kikipmtgf** [comment]
Hi,

Thanks a lot for this update! I confirm it works perfectly well with Crno-bijeli svijet. I spent some time looking for other video contents with subtitles on HRTi. It seems that there are very few of them but I managed to find some: the TV series "Novine" for instance. The subtitle feature of the add-on works also absolutely fine with them.

I haven't noticed any major slowdown neither, so my overall feeling is that it is very convenient and well executed :) Thanks again, and congrats!
